packages/swap/src/utils/rpc.ts

```diff
@@ -13,30 +13,36 @@ type Result = { success: true } | { success: false; reason: string };
 export const validateSwapAmount = async (
   asset: UncheckedAssetAndChain,
   amount: bigint,
+  blockHash?: string,
 ): Promise<Result> => {
-  const env = await cachedGetEnvironment({
-    rpcUrl: process.env.RPC_NODE_HTTP_URL as string,
-  });
+  const env = await cachedGetEnvironment(
+    { rpcUrl: process.env.RPC_NODE_HTTP_URL as string },
+    [blockHash],
+  );
 
   return validateAmount(env, asset, amount);
 };
 
 export const getIngressFee = async (
   asset: UncheckedAssetAndChain,
+  blockHash?: string,
 ): Promise<bigint> => {
-  const env = await cachedGetEnvironment({
-    rpcUrl: process.env.RPC_NODE_HTTP_URL as string,
-  });
+  const env = await cachedGetEnvironment(
+    { rpcUrl: process.env.RPC_NODE_HTTP_URL as string },
+    [blockHash],
+  );
 
   return readAssetValue(env.ingressEgress.ingressFees, asset);
 };
 
 export const getEgressFee = async (
   asset: UncheckedAssetAndChain,
+  blockHash?: string,
 ): Promise<bigint> => {
-  const env = await cachedGetEnvironment({
-    rpcUrl: process.env.RPC_NODE_HTTP_URL as string,
-  });
+  const env = await cachedGetEnvironment(
+    { rpcUrl: process.env.RPC_NODE_HTTP_URL as string },
+    [blockHash],
```

Comment:
im afraid this will break indexing swaps before the upgrade because the returned value wont be valid 😵‍💫